### PR TITLE
return proper http status failures from get logs

### PR DESCRIPTION
**Pull request by bparees**
fixes https://github.com/openshift/origin/issues/2935

### CODE REVIEW

**Comment by bparees** [comment]
@smarterclayton ptal, if you're ok with this i can look into fixing up our other rest apis.

File: pkg/build/registry/buildlog/rest.go [hunk 1]
```diff
@@ -84,11 +84,11 @@ func (r *REST) Get(ctx kapi.Context, name string, opts runtime.Object) (runtime.
 
 	// The build was cancelled
 	case api.BuildStatusCancelled:
-		return nil, fmt.Errorf("Build %s/%s was cancelled", build.Namespace, build.Name)
+		return nil, errors.NewBadRequest(fmt.Sprintf("Build %s/%s was cancelled", build.Namespace, build.Name))
```
Comment (smarterclayton) [inline pkg/build/registry/buildlog/rest.go]: lower case - `Build` should be `build` since this will be written like "error: build a/b was cancelled"

**Comment by smarterclayton** [comment]
Approved

File: pkg/build/registry/buildlog/rest.go (hunk 1 above)
Comment (liggitt) [inline pkg/build/registry/buildlog/rest.go]: We don't try to use API object names?

**Comment by bparees** [comment]
@smarterclayton lowercased and [merge]ing on your approval.

File: pkg/build/registry/buildlog/rest.go (hunk 1 above)
Comment (smarterclayton) [inline pkg/build/registry/buildlog/rest.go]: Not in bad request.

----- Original Message -----

**Pull request merged**